src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRequiredAttributeDescriptorBuilder.cs

```diff
@@ -26,6 +26,8 @@ public DefaultRequiredAttributeDescriptorBuilder(DefaultTagMatchingRuleDescripto
 
         public override RequiredAttributeDescriptor.ValueComparisonMode ValueComparisonMode { get; set; }
 
+        public override bool HasDefaultValue { get; set; }
```

Comment:
Is there a way we can make this work with just this? Aka, if something has a default value update the logic in the generated view component to construct a valid anonymous object that doesn't provide the attribute?

Reply:
> Aka, if something has a default value update the logic in the generated view component to construct a valid anonymous object that doesn't provide the attribute?

I assume that you are referring to the logic here:

https://github.com/dotnet/aspnetcore/blob/af96abad0112a90110c0f2d46104fb2e06ed2de2/src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/ViewComponentTagHelperTargetExtension.cs#L161-L169

If so, omitting the optional parameter from L165 won't do the trick since that will break the view component for scenarios where the parameter _is_ provided.

Another option is to set the default value when setting up the initialized view component itself ala

```csharp
_someInitializedViewComponent.someParameterOnItThatHasADefaultValue = "TheDefaultValue";
```

But I think that's harder to grok than setting the default value in the property definition in the class.

In either case, I think we need a way to communicate to the code-writing phase that a default value exists and I don't see a way of doing that outside of adding a new definition in the descriptor.

Reply:
I think my comment above may have been a little confusing after re-reading it now, lol I don't think I entirely understand its reasoning either 😄.

I thinkkkkkk what I may have been getting at is, what if you didn't omit the `RequiredAttributeDescriptor`'s at all for optional parameters and instead used the lacking of a required attribute descriptor on VCTH's as an indication that the attribute has a default. Once you know the attribute h as a default you can do things like:
```C#
public string foo { get; set; }
        public override void Process(TagHelperContext context, TagHelperOutput output)
        {
            var parameters = new Dictionary<string, object>();
            if (context.AllAttributes.ContainsName("foo"))
            {
                parameters["foo"] = foo;
            }
            ....
        }
```